main/lsp/tools/make_lsp_types.cc

```diff
@@ -1422,7 +1424,7 @@ void makeLSPTypes(vector<shared_ptr<JSONClassType>> &enumTypes, vector<shared_pt
             {"textDocument/formatting", makeVariant({JSONNull, makeArray(TextEdit)})},
             // (CodeAction | Command)[] | null
             // Sorbet only sends CodeAction[].
-            {"textDocument/codeAction", makeVariant({JSONNull, makeArray(CodeAction)})},
+            {"textDocument/codeAction", makeVariant({JSONNull, makeArray(CodeAction), makeArray(Command)})},
```

Comment:
The comment above and the LSP docs make it seem like the type should be

```cpp
makeVariant({JSONNull, makeArray(makeVariant({CodeAction, Command}))})
```

btw

Reply:
That was my first thought too! But turns out you can't have two json objects in the same union with our current setup:
https://github.com/sorbet/sorbet/blob/dd0e616ca5457e2b0c6e22f665e16e52022590bb/main/lsp/tools/generate_lsp_messages.h#L948-L951

Currently that version fails to compile, because `CodeAction` and `Command`  have the same `CPPBaseKind` — `BaseKind::ObjectKind`. 

I think there is a way around this, the `CodeAction` class has a optional `command` field, we can use that. I'm testing this hypothesis rn